### ISSUE TITLE
Question about using mel spectrogram mask for text inputs

### ISSUE THREAD

**Issue by JosefAlbers** [lucidrains/e2-tts-pytorch#38]
I discovered your repository a few weeks ago and have become a big fan.

While going through the code, I noticed something interesting on line [707](https://github.com/lucidrains/e2-tts-pytorch/blob/195ef0db35d7a30499546980e92204f68cf79ac4/e2_tts_pytorch/e2_tts.py#L707), where the mask generated from mel spectrograms is used also for texts. In my own simplified version, I found that not using the mask for text seemed to work better.

I’d love to learn of your thoughts on why the mel spectrogram mask is used for text. What benefits does it offer that I might be missing?

**Comment by lucasnewman** [comment]
I would guess it would be nearly equivalent but not really necessary, since the model is enforcing that the text length is strictly less than the number of audio frames. This is briefly mentioned in the paper:

<img width="456" alt="Screenshot 2024-10-15 at 11 57 32 AM" src="https://github.com/user-attachments/assets/9ad66d0f-6a26-4b13-8a0e-f6cfb68e9cc5">


It's probably improves training efficiency without the mask, though! Are you seeing a significant difference?

**Comment by JosefAlbers** [comment]
Oh, I see! Thanks so much, @lucasnewman! I initially thought removing the mask led to dramatic improvements in training loss, but after several comparison runs, I’m not seeing much difference. What I attributed to the mask removal may have actually been due to other changes I made in my code. Sorry for the mix-up, and thanks again for setting me straight!
